test/hotspot/jtreg/compiler/loopopts/superword/TestMulAddS2I.java

```diff
@@ -91,4 +97,40 @@ public static int[] test() {
         }
         return out;
     }
+
+    @Test
+    @IR(applyIfCPUFeature = {"sse2", "true"},
+        applyIfPlatform = {"64-bit", "true"},
+        counts = {IRNode.MUL_ADD_S2I, "> 0", IRNode.MUL_ADD_VS2VI, "> 0"})
+    @IR(applyIfCPUFeature = {"asimd", "true"},
+        applyIf = {"MaxVectorSize", "16"}, // AD file requires vector_length = 16
+        counts = {IRNode.MUL_ADD_S2I, "> 0", IRNode.MUL_ADD_VS2VI, "> 0"})
+    @IR(applyIfCPUFeature = {"avx512_vnni", "true"},
+        counts = {IRNode.MUL_ADD_S2I, "> 0", IRNode.MUL_ADD_VS2VI_VNNI, "> 0"})
+    public static int[] testb() {
+        int[] out = new int[ITER];
+        int[] out2 = new int[ITER];
+        for (int i = 0; i < ITER; i++) {
+            out[i] += ((sArr1[2*i] * sArr2[2*i]) + (sArr1[2*i+1] * sArr2[2*i+1]));
```

Comment:
Note: `testa` only uses `sArr1`, and I wanted to be sure it works also with this.